### DISCUSSION TITLE
Should we move DbContextCheck from Web to Infra?

### DISCUSSION THREAD

**Discussion by tmhdev01** [jasontaylordev/CleanArchitecture#1157] [Q&A]
Hi all,

I am starting a project based on this repo. It's an enjoyable journey thanks for your great job!

There is a point that I am concerned about: Should we move DbContextCheck from Web to Infra? From my PoV, it is the specific service registering for the context class (ApplicationDbContext), not for the interface (IApplicationDbContext). It seems to be more "clean" to be placed in the Infra project.

What is your view? Feel free to share your comments. I already have the PR if you guys like it :)

![image](https://github.com/jasontaylordev/CleanArchitecture/assets/1224095/4efdb5f4-b089-4171-a7c7-25e06ae9c2b4)

**Comment by jasontaylordev** [comment]
Hi @tmhdev01 - I'm glad you're enjoying the template. 😀

This code relates to [ASP.NET Core Health Checks](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks?view=aspnetcore-8.0). `AddHealthChecks` registers health check services, and `AddDbContextCheck` adds a health check for the specified DbContext type.

Since this code relates to ASP.NET Core Health Checks, moving part of it to Infrastructure would make it more difficult to understand how health checks are configured. Regardless, it's a good idea to try it in your solutions and see if you prefer the approach.

I'm open to further discussion and always happy to discuss changes that might simplify or improve the template somehow.
